Dockerfile

```diff
@@ -34,3 +34,8 @@ RUN yum update -y \
   && rm stack.tgz \
   && mv stack /usr/local/bin/ \
   && cd .. && rm -r /tmp/stack
+
+ARG USER=haskell
+RUN useradd --create-home "$USER"
+RUN echo "$USER ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
```

Comment:
These could be moved into the one giant `RUN` command as well. Gotta keep the number of layers down! 😆

Reply:
Does keeping the number of layers low reduce file-size? At least in nix, more layers enable more sharing and reduce file-size:

https://grahamc.com/blog/nix-and-layered-docker-images

It'd be nice to know if the opposite is true for docker. Maybe nix automatically uses a union filesystem like AUFS or something.

Reply:
Each layer produces mostly minimal overhead, the issue is that a snapshot is created for each one so if you do something like:

```docker
RUN wget a-2gb-file.zip
RUN rm a-2gb-file.zip
```
your image size would still be about 2gb (plus everything else).